internal/hvsocket/hvsocket.go

```diff
@@ -0,0 +1,91 @@
+//go:build windows
+// +build windows
+
+package hvsocket
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"unicode/utf16"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"golang.org/x/sys/windows"
+)
+
+func HCSIDToGUID(id string) (guid.GUID, error) {
+	var buf bytes.Buffer
+	if err := binary.Write(&buf, binary.LittleEndian, utf16.Encode([]rune(strings.ToUpper(id)))); err != nil {
+		return guid.GUID{}, err
+	}
+	g, err := guid.NewV5(guid.GUID{Data1: 0xcab70344, Data2: 0xfacb, Data3: 0x41e4, Data4: [8]byte{0xb5, 0xe5, 0xab, 0x65, 0x92, 0x28, 0x3e, 0x6e}}, buf.Bytes())
+	if err != nil {
+		return guid.GUID{}, err
+	}
+	return g, nil
+}
+
+const (
+	cAddressFlagPassthru            = 0x00000001
+	cIOCtlHVSocketUpdateAddressInfo = 0x21c004
+)
+
+type tAddressInfo struct {
+	SystemID         guid.GUID
+	VirtualMachineID guid.GUID
+	SiloID           guid.GUID
+	Flags            uint32
+}
```

Comment:
I think you've been looking at C++ code for too long 😅
```suggestion
const (
	addressFlagPassthru            = 0x00000001
	ioCtlHVSocketUpdateAddressInfo = 0x21c004
)

type addressInfo struct {
	systemID         guid.GUID
	virtualMachineID guid.GUID
	siloID           guid.GUID
	flags            uint32
}
```

Reply:
That's Kevin, not me 😄